src/test/java/hudson/scm/IntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+package hudson.scm;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import static org.junit.Assert.*;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class IntegrationTest {
+
+    private static final Logger LOGGER = Logger.getLogger(IntegrationTest.class.getName());
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+    @Rule public TemporaryFolder tmp = new TemporaryFolder();
+    private File repo;
+    private ServerSocket sock;
+    private File work;
+
+    private static ProcessBuilder inheritIO(ProcessBuilder pb) throws Exception {
+        try {
+            ProcessBuilder.class.getMethod("inheritIO").invoke(pb);
+        } catch (NoSuchMethodException x) {
+            // Java 6, ignore
```

Comment:
Because it is not an error condition; running tests using Java 6 is still supported.